apps/projects/dashboard.py

```diff
@@ -61,5 +64,45 @@ def get_urls(self):
         ]
 
 
+class InsightForm(ProjectDashboardForm):
+    class Meta:
+        model = Insight
+        fields = [
+            "active_participants",
+            "comments",
+            "ratings",
+            "written_ideas",
+            "poll_answers",
+            "live_questions",
+            "display",
+        ]
+        required_for_project_publish = []
+
+        help_texts = {
+            "display": _(
+                "Show insights with numbers "
+                "of contributions and participants "
+                "during the participation process"
+            ),
+        }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        for name, field in self.fields.items():
+            if not name == "display":
+                field.widget.attrs["readonly"] = "true"
+
+
+class ProjectInsightComponent(ProjectFormComponent):
+    identifier = "insight"
+    weight = 12
+    label = _("Insights")
+
+    form_title = _("Insights")
+    form_class = InsightForm
+    form_template_name = "a4_candy_projects/includes/project_insight_form.html"
```

Comment:
@Rineee I follow the logic of a4 with the templates of components, but in this case, we want the insights to be part of result template. did u have to do that before, am a lil lost with all the templates in a4. If not, I need to look into it further.
```
class ProjectResultComponent(ProjectFormComponent):
    identifier = "result"
    weight = 12
    label = _("Result")

    form_title = _("Edit project result")
    form_class = forms.ProjectResultForm
    form_template_name = "a4dashboard/includes/project_result_form.html"
```
ref: https://github.com/liqd/adhocracy4/blob/main/adhocracy4/dashboard/dashboard.py#L30C1-L37C73

Or any other suggestion?

Reply:
It works for a component, but when I extend with this the result form component, is not showing up anything @Rineee

Reply:
@m4ra Here is an example where we overwrote a dashboard component lately: https://github.com/liqd/a4-meinberlin/pull/5095/commits
I am not sure if you need to overwrite completely or it would also work to extend the component by just overwriting the `get_urls()`? But I think you need to name it ProjectResultComponent also and then do `components.replace_project(ProjectResultComponent())`.